dlproject/data/mnist_dataset.py

```diff
@@ -13,13 +13,13 @@ class MNISTDatasetBuilder:
     def __init__(self):
         (self.__train_x, self.__train_y), (self.__test_x, self.__test_y) = tf.keras.datasets.mnist.load_data()
         
-    def preprocess_dataset_simple_ae(self):
+    def preprocess_dataset_simple_ae(self, noise_factor: float = 0.4):
         self.__train_x = self.__preprocess_array(self.__train_x)
         self.__test_x = self.__preprocess_array(self.__test_x)
-        self.__noisy_train_data = self.__add_noise(self.__train_x)
-        self.__noisy_test_data = self.__add_noise(self.__test_x)
+        self.__noisy_train_data = self.__add_noise(self.__train_x, noise_factor)
+        self.__noisy_test_data = self.__add_noise(self.__test_x, noise_factor)
 
-    def preprocess_dataset_simple_vae(self):
+    def preprocess_dataset_simple_vae(self, noise_factor: float = 0.4):
```

Comment:
Perché aggiungerlo anche qua se non viene usato?

Reply:
Credo che servirà quando verrà aggiunto il rumore in preprocess_dataset_simple_vae()